group-coordinator/src/main/java/org/apache/kafka/coordinator/group/ShareGroupAutoOffsetResetStrategy.java

```diff
@@ -0,0 +1,155 @@
+
+        StrategyType shareGroupType;
+        switch (baseType) {
+            case EARLIEST:
+                shareGroupType = StrategyType.EARLIEST;
+                break;
+            case LATEST:
+                shareGroupType = StrategyType.LATEST;
+                break;
+            case BY_DURATION:
+                shareGroupType = StrategyType.BY_DURATION;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported strategy for ShareGroup: " + baseType);
+        }
+
+        return new ShareGroupAutoOffsetResetStrategy(baseStrategy, shareGroupType);
+    }
+
+    /**
+     * Returns the share group strategy type.
+     */
+    public StrategyType type() {
+        return type;
+    }
+
+    /**
+     * Returns the name of the share group offset reset strategy.
+     */
+    public String name() {
+        return type.toString();
+    }
+
+    /**
+     * Delegates the timestamp calculation to the base strategy.
+     * @return the timestamp for the OffsetResetStrategy,
+     * if the strategy is EARLIEST or LATEST or duration is provided
+     * else return Optional.empty()
+     */
+    public Long timestamp() {
+        return delegate.timestamp().get();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ShareGroupAutoOffsetResetStrategy that = (ShareGroupAutoOffsetResetStrategy) o;
+        return type == that.type && Objects.equals(delegate, that.delegate);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(delegate, type);
+    }
+
+    @Override
+    public String toString() {
+        return "ShareGroupAutoOffsetResetStrategy{" +
+                "type=" + type +
+                ", delegate=" + delegate +
+                '}';
+    }
+
+    /**
+     * Factory method for creating EARLIEST strategy.
+     */
+    public static ShareGroupAutoOffsetResetStrategy earliest() {
+        return new ShareGroupAutoOffsetResetStrategy(AutoOffsetResetStrategy.EARLIEST, StrategyType.EARLIEST);
+    }
+
+    /**
+     * Factory method for creating LATEST strategy.
+     */
+    public static ShareGroupAutoOffsetResetStrategy latest() {
+        return new ShareGroupAutoOffsetResetStrategy(AutoOffsetResetStrategy.LATEST, StrategyType.LATEST);
+    }
+
+    public static class Validator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            String offsetStrategy = (String) value;
+            try {
+                fromString(offsetStrategy);
+            } catch (Exception e) {
+                throw new ConfigException(name, value, "Invalid value `" + offsetStrategy + "` for configuration " +
+                        name + ". The value must be either 'earliest', 'latest' or of the format 'by_duration:<PnDTnHnMn.nS.>'.");
+            }
+        }
+    }
+}
```

Comment:
This validator needs a `toString()` method. Otherwise, the documentation will include the validator's object reference instead of a string for the valid options. I suggest:
```
   public String toString() {
            return "[earliest, latest, by_duration:PnDTnHnMn.nS]";
        }
```
That worked nicely in the generated HTML.